### PR TITLE
Universal block inspector controls back button

### PR DESCRIPTION
**Pull request by getdave**
## What?

In https://github.com/WordPress/gutenberg/pull/46037#discussion_r1031542859 we modified the back button in the inspextor controls so that it selects the closest parent Navigatino block.

This PR tries to make that solution universal to all block types outside of the experiment by seeking the first parent block that is controlling inner blocks and selecting that.

### CODE REVIEW

File: packages/block-editor/src/components/block-card/index.js [hunk 1]
```diff
@@ -30,29 +30,52 @@ function BlockCard( { title, icon, description, blockType, className } ) {
...
 		<div className={ classnames( 'block-editor-block-card', className ) }>
-			{ isOffCanvasNavigationEditorEnabled && parentNavBlockClientId && (
+			{ showBackButton && (
 				<Button
-					onClick={ () => selectBlock( parentNavBlockClientId ) }
-					label={ __( 'Go to parent Navigation block' ) }
+					onClick={ () =>
+						selectBlock( closestControllingBlock?.clientId )
+					}
+					label={ __( 'Go to controlling block' ) }
```
Comment (alexstine) [inline packages/block-editor/src/components/block-card/index.js:78]: Please make this label more explanatory. Example: `label={ sprintf( __( 'Back to parent block: %s', parentBlock.title ) ) }`

**Review by alexstine — COMMENTED** [review]
@getdave This looks reasonable to me. One blocker below.

File: packages/block-editor/src/components/block-card/index.js [hunk 2]
```diff
@@ -30,29 +30,52 @@ function BlockCard( { title, icon, description, blockType, className } ) {
...
 	const { selectBlock } = useDispatch( blockEditorStore );
 
+	// If the selected block is being controlled then show a back button which
+	// allows jumping immediately up to the controlling ancestor block.
+	// Currently limited to the offcanvas experiment but may be opened up to all
+	// blocks which meet the criteria in future.
+	const showBackButton =
+		isOffCanvasNavigationEditorEnabled &&
+		closestControllingBlock?.name !== 'core/template-part' &&
+		closestControllingBlock?.name !== 'core/block';
```
Comment (getdave) [inline packages/block-editor/src/components/block-card/index.js:69]: These values could come from block editor settings?

**Comment by getdave** [comment]
I'm not sure this PR is needed anymore so I"m closing it out.

**Pull request closed without merging**